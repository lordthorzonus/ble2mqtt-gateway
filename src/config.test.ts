jest.mock("fs");

import { readFileSync } from "fs";

const mockReadFileSync = readFileSync as jest.Mock;

const exampleConfiguration = `
mqtt:
  host: mqtt.lan.fi
  port: 8883
  username: a
  password: b
  client_id: c

gateways:
  base_topic: mqtt
  ruuvitag:
    allow_unknown: true
    devices:
      - name: fridge_ruuvitag
        id: da21045d81a8
      - name: balcony_ruuvitag
        id: 641cae0910b5
  miflora:
    devices:
      - name: other_plant
        id: c47c8d6e07cf
      - name: some_plant
        id: c47c8d6e069c

homeassistant:
  discovery_topic: homeassistant
`;

describe("Config", () => {
    afterEach(() => {
        jest.resetAllMocks();
        jest.resetModules();
    });

    afterAll(() => {
        jest.restoreAllMocks();
    });

    describe("getConfiguration()", () => {
        it("should return the proper configuration given in the env variable", () => {
            process.env.CONFIG_FILE_LOCATION = "test";
            mockReadFileSync.mockReturnValue(exampleConfiguration);
            expect(jest.requireActual("./config").getConfiguration()).toMatchSnapshot();
            expect(mockReadFileSync).toHaveBeenCalledWith("test", "utf-8");
        });
    });
});
