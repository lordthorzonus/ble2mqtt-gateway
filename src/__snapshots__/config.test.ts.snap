// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config getConfiguration() should return the proper configuration given in the env variable 1`] = `
Object {
  "gateways": Object {
    "base_topic": "mqtt",
    "miflora": Object {
      "devices": Array [
        Object {
          "id": "c47c8d6e07cf",
          "name": "Other Plant",
        },
        Object {
          "id": "c47c8d6e069c",
          "name": "some_plant",
        },
      ],
      "timeout": 60000,
    },
    "ruuvitag": Object {
      "allow_unknown": true,
      "devices": Array [
        Object {
          "id": "da21045d81a8",
          "name": "Fridge Ruuvitag",
        },
        Object {
          "id": "641cae0910b5",
          "name": "balcony_ruuvitag",
        },
      ],
      "timeout": 30000,
    },
  },
  "homeassistant": Object {
    "discovery_topic": "homeassistant",
  },
  "mqtt": Object {
    "client_id": "c",
    "host": "mqtt.lan.fi",
    "password": "b",
    "port": 8883,
    "protocol": "mqtt",
    "username": "a",
  },
}
`;
